Add exec-mode to script files

gargnitingoogle opened the pull request:
### Description
Add execute mode to all shell scripts
    
This change adds execute mode (chmod +x)
to add script files in the gcsfuse project. This is needed to avoid having to
`chmod +x` each time for running a shell
script especially a in jobs or in other scripts.
This `chmod +x` caused errors in running `git checkout`
in the calling script in the scenario like below.
    
```shell
git checkout master
chmod +x some_child_script.sh
# run some tests
git checkout dev_branch ### FAILS saying files are modified.
# run same tests
```
    
This change addresses internal bug#307698513 .

This has a follow-up cleanup in [Remove chmod calls for running scripts](https://github.com/GoogleCloudPlatform/gcsfuse/pull/1489) . Currently these two changes can't be merged together because putting them together fails the presubmit tests.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - All run locally and passed.
3. Integration tests - Ran through presubmits

ashmeenkaur reviewed (commented):
KOKORO tests are failing. Is this expected?

gargnitingoogle commented:
> KOKORO tests are failing. Is this expected?

No, I'll look at the failure.

gargnitingoogle commented:
> > KOKORO tests are failing. Is this expected?
> 
> No, I'll look at the failure.

The failure was because I had also removed the chmod calls also within the same change, which caused the presubmits to fail. The presubmit jobs do the following:

```shell
git checkout master
chmod +x run_some_script.sh # Earlier moved in this PR itself.

./run_some_script.sh
# Fails because even though we have add exec-permission on run_some_script.sh in this PR/branch, it's not reflected on the master branch until the merge goes through, and master branch doesn't have run_some_script.sh as executable.

# verify output
git checkout dev-branch # FAILS 
./run_some_script.sh
```

I've addressed this failure by splitting out the chmod removals into a cleanup PR afterwards i.e. [Remove chmod calls for running scripts](https://github.com/GoogleCloudPlatform/gcsfuse/pull/1489).

gargnitingoogle commented:
This PR is safe to be merged into the release branch along the same line as [here](https://github.com/GoogleCloudPlatform/gcsfuse/pull/1450#issuecomment-1788729113) .

ashmeenkaur commented on perfmetrics/scripts/presubmit_test/pr_perf_test/build.sh:
```diff
@@ -90,6 +90,7 @@ if [[ "$perfTestStr" == *"$EXECUTE_PERF_TEST_LABEL"* ]];
 then
  # Executing perf tests for master branch
  install_requirements
+ git reset --hard
```
Remove git reset --hard before merging

raj-prince commented:
I can see there are 21 shell-scripts under `perfmetrics/script`. Do we have any reason to leave some of them?
E.g., ml_tests/pytorch/dino/setup_container.sh

gargnitingoogle commented:
> I can see there are 21 shell-scripts under `perfmetrics/scripts`. Do we have any reason to leave some of them? E.g., ml_tests/pytorch/dino/setup_container.sh is not present in this PR.

I haven't done it blanketly for all scripts, only for those scripts that were being `chmod`ed somewhere or the other.

raj-prince commented:
> I haven't done it blanketly for all scripts, only for those scripts that were being `chmod`ed somewhere or the other.

Got it. Confused with the PR description `Add execute mode to all shell scripts` , we can update that.

gargnitingoogle commented:
The presubmit tests are all passing now. [link](https://fusion2.corp.google.com/ci/kokoro/prod:gcsfuse%2Fgcp_ubuntu%2Fpresubmits%2Fperf_tests%2Fpresubmit/activity/deb25863-e46a-4b9e-9d23-a5f59c3f92bc/log)

Perf results
+--------+------------+--------------+------------+--------------+--------------+
| Branch | File Size  |   Read BW    |  Write BW  | RandRead BW  | RandWrite BW |
+--------+------------+--------------+------------+--------------+--------------+
| Master |  0.25MiB   | 505.41MiB/s  | 1.04MiB/s  |  35.39MiB/s  |  1.08MiB/s   |
|   PR   |  0.25MiB   | 498.23MiB/s  | 0.98MiB/s  |  35.91MiB/s  |  1.09MiB/s   |
|        |            |              |            |              |              |
|        |            |              |            |              |              |
| Master | 48.828MiB  | 4807.47MiB/s | 80.9MiB/s  | 1219.08MiB/s |  79.66MiB/s  |
|   PR   | 48.828MiB  | 4787.23MiB/s | 73.52MiB/s | 1039.77MiB/s |  80.18MiB/s  |
|        |            |              |            |              |              |
|        |            |              |            |              |              |
| Master | 976.562MiB | 4784.92MiB/s | 32.62MiB/s | 448.63MiB/s  |  32.0MiB/s   |
|   PR   | 976.562MiB | 4857.06MiB/s | 32.83MiB/s | 539.19MiB/s  |  32.15MiB/s  |
|        |            |              |            |              |              |
|        |            |              |            |              |              |
+--------+------------+--------------+------------+--------------+--------------+

gargnitingoogle commented:
@ashmeenkaur @raj-prince I'm have added `git reset --hard` calls to avoid breaking presubmits in the meantime before the 2nd PR goes in. The 2nd PR undoes these git reset --hard calls. 

Also earlier I had only added the exec-permissions to those script files, which were being `chmod`ed somewhere in the code, but I've now extended the change to all the script files to match the PR title and description.

Please review and approve.
